### PR TITLE
Add csr manager for newgemm

### PR DESCRIPTION
**Pull request by xiaoling-yi**
This RP,
* adds the csrManager for new gemm to be integrated with streamer. 
* give the new gemm the same interface to SNAX as streamer (the csrReqRspIO).

As shown in the figure below, 

image

The source code is the same as csrManager in https://github.com/KULeuven-MICAS/snax-streamer/blob/main/src/main/scala/streamer/CsrManager.scala

### CODE REVIEW

File: src/main/scala/gemm/BareBlockGemm.scala [hunk 1]
```diff
@@ -189,3 +189,37 @@ object BareBlockGemm extends App {
...
+
+  bareBlockGemm.io.ctrl.valid := csrManager.io.csr_config_out.valid
+  csrManager.io.csr_config_out.ready := bareBlockGemm.io.ctrl.ready
+  bareBlockGemm.io.ctrl.bits.M_i := csrManager.io.csr_config_out.bits(0)
+  bareBlockGemm.io.ctrl.bits.K_i := csrManager.io.csr_config_out.bits(1)
+  bareBlockGemm.io.ctrl.bits.N_i := csrManager.io.csr_config_out.bits(2)
+  bareBlockGemm.io.ctrl.bits.subtraction_a_i :=
+    csrManager.io.csr_config_out.bits(3)
+  bareBlockGemm.io.ctrl.bits.subtraction_b_i :=
+    csrManager.io.csr_config_out.bits(4)
```
Comment (rgantonio) [inline src/main/scala/gemm/BareBlockGemm.scala:214]: Add commented documentation on what these parameters are 😄 like what you did for others!

File: src/main/scala/gemm/CsrManager.scala
Comment (rgantonio) [inline src/main/scala/gemm/CsrManager.scala:1]: The realization that the CSR manager is actually "similar" to the one in SNAX streamer. Not the same but similar looking. I did not foresee this before but I guess this is fine haha

**Review by rgantonio — APPROVED** [review]
Approved! Please check comment about which parameters to add 😄

File: src/main/scala/gemm/BareBlockGemm.scala (hunk 1 above)
Comment (xiaoling-yi) [reply src/main/scala/gemm/BareBlockGemm.scala:214]: ok!

**Pull request merged**